src/odemis/driver/smaract.py

```diff
@@ -1814,7 +1824,19 @@ def SA_MC_GetProperty_s(self, id, property_key, val, ioArraySize):
 
     def SA_MC_WaitForEvent(self, id, p_ev, timeout):
         ev = _deref(p_ev, SA_MC_Event)
-        time.sleep(0.25)
+        
+        # Simulate the intermediate position of the move
+        n_steps = 5
+        for i in range(n_steps):
+            intermediate = SA_MC_Pose()
+            intermediate.x = self.pose.x + (self.target.x - self.pose.x) * i / n_steps
+            intermediate.y = self.pose.y + (self.target.y - self.pose.y) * i / n_steps
+            intermediate.z = self.pose.z + (self.target.z - self.pose.z) * i / n_steps
+            intermediate.rx = self.pose.rx + (self.target.rx - self.pose.rx) * i / n_steps
+            intermediate.rz = self.pose.rz + (self.target.rz - self.pose.rz) * i / n_steps
+            self.pose = copy.copy(intermediate)
+            time.sleep(0.05)
+        
```

Comment:
That doesn't properly simulate the move! You have to take into account the speed. I'd highly suggest to not put the simulation code in the WaitForEvent(), but in the GetPose(). See in the TMCM simulator how this is done.